### PR TITLE
Accent folded search

### PR DESCRIPTION
**Pull request by bryvin**
This replaces accented chars when searching notes so it matches on the non-accented versions of everything. 

Found a list from online, not sure if there are any missing accented but this list seems pretty consistent.

### CODE REVIEW

File: app/assets/javascripts/app/controllers/notes.js [hunk 1]
```diff
@@ -246,46 +246,57 @@ angular.module('app')
...
-        ["[ìíîï]", "i"],
-        ["ñ", "n"],
-        ["[òóôõö]", "o"],
-        ["œ", "oe"],
-        ["[ùúûü]", "u"],
-        ["[ýÿ]", "y"],
-        ["\\W", ""]
+        [/ae/g, "(?:æ|ae)"], // special because 2 characters => only 1
+        [/oe/g, "(?:œ|oe)"], // special because 2 characters => only 1
+        [/a/g, "[aàáâãäå]"],
```
Comment (tjrana) [inline app/assets/javascripts/app/controllers/notes.js]: Implement ā

File: app/assets/javascripts/app/controllers/notes.js [hunk 2]
```diff
@@ -246,46 +246,57 @@ angular.module('app')
...
+        [/e/g, "[eèéêë]"],
```
Comment (tjrana) [inline app/assets/javascripts/app/controllers/notes.js]: Implement ēėę

File: app/assets/javascripts/app/controllers/notes.js [hunk 3]
```diff
@@ -246,46 +246,57 @@ angular.module('app')
...
+        [/i/g, "[iìíîï]"],
```
Comment (tjrana) [inline app/assets/javascripts/app/controllers/notes.js]: Implement īį

File: app/assets/javascripts/app/controllers/notes.js [hunk 4]
```diff
@@ -246,46 +246,57 @@ angular.module('app')
...
+        [/o/g, "[oòóôõö]"],
```
Comment (tjrana) [inline app/assets/javascripts/app/controllers/notes.js]: Implement øō

File: app/assets/javascripts/app/controllers/notes.js [hunk 5]
```diff
@@ -246,46 +246,57 @@ angular.module('app')
...
+        [/u/g, "[uùúûü]"],
```
Comment (tjrana) [inline app/assets/javascripts/app/controllers/notes.js]: Implement ū

File: app/assets/javascripts/app/controllers/notes.js [hunk 6]
```diff
@@ -246,46 +246,57 @@ angular.module('app')
...
+        [/y/g, "[yýÿ]"],
+        [/c/g, "[cç]"],
```
Comment (tjrana) [inline app/assets/javascripts/app/controllers/notes.js]: Implement ćč

File: app/assets/javascripts/app/controllers/notes.js [hunk 7]
```diff
@@ -246,46 +246,57 @@ angular.module('app')
...
+        [/n/g, "[nñ]"]
```
Comment (tjrana) [inline app/assets/javascripts/app/controllers/notes.js]: Implement ń

**Comment by tjrana** [comment]
The following characters still need to be implemented:
Implement
- [ā](https://github.com/standardnotes/web/pull/246/commits/c04e875eb78bdab6594b11ff3a9919c76fd44c93#r224160755)
- [ēėę](https://github.com/standardnotes/web/pull/246/commits/c04e875eb78bdab6594b11ff3a9919c76fd44c93#r224161034)
- [īį](https://github.com/standardnotes/web/pull/246/commits/c04e875eb78bdab6594b11ff3a9919c76fd44c93#r224161244)
- [øō](https://github.com/standardnotes/web/pull/246/commits/c04e875eb78bdab6594b11ff3a9919c76fd44c93#r224161491)
- [ū](https://github.com/standardnotes/web/pull/246/commits/c04e875eb78bdab6594b11ff3a9919c76fd44c93#r224161691)
- [ćč](https://github.com/standardnotes/web/pull/246/commits/c04e875eb78bdab6594b11ff3a9919c76fd44c93#r224161945)
- [ń](https://github.com/standardnotes/web/pull/246/commits/c04e875eb78bdab6594b11ff3a9919c76fd44c93#r224162050)
- many more...

**Comment by tjrana** [comment]
Example implementation:

```js
const regex = /(?=[\s\S]*f[oôöòóøōõ][oôöòóøōõ])(?=[\s\S]*b[aàáâäãåā])[\s\S]+/gmi;
const str = `foo bar`;
let m;

while ((m = regex.exec(str)) !== null) {
  // Avoid zero-width matches infinite loops
  if (m.index === regex.lastIndex) {
    regex.lastIndex++;
  }
  // Access result with `m`
  m.forEach((match, groupIndex) => {
    console.log(`Found match, group ${groupIndex}: ${match}`);
  });
}
```

File: app/assets/javascripts/app/controllers/notes.js (hunk 3 above)
Comment (bryvin) [reply app/assets/javascripts/app/controllers/notes.js]: What is this letter combo meant to be equal to?

**Comment by mobitar** [comment]
After further review, we have decided to put this PR on hold for mainly two reasons:

- in Chrome, with the Developer Tools open, performance is really bad. Noticeably bad. However, strangely, this doesn't happen with any other browser. 
- It doesn't handle the case where your input has an accented character. (Easy fix probs)

**Pull request closed without merging**